packages/programs/src/programs/ForceFieldProgram.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { IBaseWorld } from "@latticexyz/world-consumer/src/experimental/WorldConsumer.sol";
+
+import { EntityId } from "@dust/world/src/EntityId.sol";
+import { ObjectTypeId } from "@dust/world/src/ObjectTypeId.sol";
+import { ObjectAmount } from "@dust/world/src/ObjectTypeLib.sol";
+import { ProgramId } from "@dust/world/src/ProgramId.sol";
+import { Vec3 } from "@dust/world/src/Vec3.sol";
+
+import { ReversePlayer } from "@dust/world/src/codegen/tables/ReversePlayer.sol";
+
+import {
+  IAddFragmentHook,
+  IBuildHook,
+  IMineHook,
+  IProgramValidator,
+  IRemoveFragmentHook
+} from "@dust/world/src/ProgramInterfaces.sol";
+
+import { Admin } from "../codegen/tables/Admin.sol";
+import { AllowedPlayers } from "../codegen/tables/AllowedPlayers.sol";
+import { AllowedPrograms } from "../codegen/tables/AllowedPrograms.sol";
+import { DefaultPrograms } from "../codegen/tables/DefaultPrograms.sol";
+
+import { DefaultProgram } from "./DefaultProgram.sol";
+
+/**
+ * @title ForceFieldProgram
+ */
+contract ForceFieldProgram is
+  DefaultProgram,
+  IProgramValidator,
+  IAddFragmentHook,
+  IRemoveFragmentHook,
+  IBuildHook,
+  IMineHook
+{
+  /**
+   * @notice Initializes the ForceFieldProgram
+   * @param _world The world contract
+   */
+  constructor(IBaseWorld _world) DefaultProgram(_world) { }
+
+  function validateProgram(
+    EntityId caller,
+    EntityId target,
+    EntityId programmed,
+    ProgramId program,
+    bytes memory extraData
+  ) external view onlyWorld {
+    bytes32[] memory defaultPrograms = DefaultPrograms.get();
+    for (uint256 i = 0; i < defaultPrograms.length; i++) {
+      if (defaultPrograms[i] == program.unwrap()) {
```

Comment:
I see now the need for default programs table, but this one is kind of weird, I don't think the default forcefield should just allow default programs, a saner default imo is to accept any program

Reply:
and maybe allow restricting to specific programs with another function

Reply:
for now removing restricting all together. default can just approve any program yea